Ingested Gspread Data Into Postgres Database and Utitlized Dbeaver to write SQL Scripts

Humphery7 opened the pull request:
@elseagle 
@tobaadesugba 
@AmazingGrace-D 

-> Created Postgres Database
-> Utilized gspread library to get data from google spreadsheet  
-> Utilized psycopg2 library to send sql queries to database to perform actions(create_table, truncate_table, insert_into_table...)
-> Installed Dbeaver and connection setup to write sql scripts to database
-> Managed Packages with poetry
-> Validated Incoming data
-> logged errors encounted with logging library

LogicAL007 committed:
Data ingestion scripts

LogicAL007 committed:
Added docstring to the functions

LogicAL007 committed:
Create README.md

LogicAL007 committed:
Update README.md

Humphery7 committed:
ingested data from google_spreadsheet to postgre database

Humphery7 committed:
added entry point

Humphery7 committed:
utiized if__name__==__main__

LogicAL007 committed:
included the data validation in the code

this validates the data type of the excel sheet

LogicAL007 committed:
Merge branch 'data_ingestion' of https://github.com/Data-Epic/Ghost_Warriors into data_ingestion

elseagle commented on Project5/project5/app.py:
```diff
@@ -0,0 +1,114 @@
+            }
+            validated_data.append(valid_record)
+        except (ValueError, KeyError) as e:
+            print(f"Skipping row due to error: {e}")
```
Do not use print, use the logging library

elseagle commented on Project5/project5/app.py:
```diff
@@ -0,0 +1,114 @@
+                insert_values_sql, values
+            )
+        conn.commit()
+        print("Data inserted successfully")
```
Kindly remove all prints and use logging if needed

elseagle commented on Project5/project5/app.py:
```diff
@@ -0,0 +1,114 @@
+                record['longitude'],
+                record['timestamp'] if record['timestamp'] else None
+            )
+            with open('Project5/project5/insert_values.sql', 'r') as file:
```
This belongs to .env

elseagle commented on Project5/project5/config.py:
```diff
@@ -0,0 +1,19 @@
+from configparser import ConfigParser
+
+
+def config(filename='Project5/project5/database.ini', section='postgresql'):
```
Magic string here should be an environment variable

elseagle commented on Project5/project5/create_table.sql:
```diff
@@ -0,0 +1,12 @@
+    state VARCHAR(100) NOT NULL,
+    country VARCHAR(100) NOT NULL,
+    wind_direction VARCHAR(100) NOT NULL,
+    temp_c INT NOT NULL,
```
What's temp_c? Not quite descriptive

elseagle replied:
This is yet to be resolved @Humphery7 @LogicAL007

elseagle commented on Project5/pyproject.toml:
```diff
@@ -0,0 +1,20 @@
+name = "project5"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Your names should be here ¯\_(ツ)_/¯

elseagle commented on Project5_II/python_sql.py:
```diff
@@ -0,0 +1,96 @@
+    #get password from environment variable
+    PASSWORD = os.getenv("PASSWORD")
+    #connect to postgres database
+    with psycopg2.connect(host="localhost",dbname="GspreadData_Storage",
```
Your db name needs to look better. You can stick with lower case and use a descriptive name - `gspreaddata_storage` doesn't sound good. You should name your db based on the data, use case or task (I mean what is being achieved)

elseagle commented on Project5_II/python_sql.py:
```diff
@@ -0,0 +1,96 @@
+
+            #sql function to crreate table if it doesn't exist
+            createTable = """CREATE TABLE IF NOT EXISTS Github_Records(
+                ID SERIAL NOT NULL,
```
Not a good naming, stick to lower case

elseagle commented on Project5_II/python_sql.py:
```diff
@@ -0,0 +1,96 @@
+                Repository_Name VARCHAR(100) NOT NULL,
+                Language_Used VARCHAR(50),
+                Description VARCHAR(200),
+                Datetime_Posted VARCHAR(50) NOT NULL,
```
`created_at` sounds better.

elseagle reviewed (changes requested):
Good work. Thank you.

I noticed you only have 1 PR, the instruction says otherwise. Kindly ensure you give more attention to details in these scenarios. All details matter.
<img width="847" alt="Screenshot 2023-11-25 at 15 36 10" src="https://github.com/Data-Epic/Ghost_Warriors/assets/27138575/34c6bcd2-b788-47f1-a896-c1946fa0a65c">

Kindly attend to requested changes too

Humphery7 committed:
maded changes as requested in review

Humphery7 committed:
Merge branch 'data_ingestion' of https://github.com/Data-Epic/Ghost_Warriors into data_ingestion

Humphery7 reviewed (commented):
@elseagle 
@tobaadesugba 

-> created better names for database and table 
-> names are ensured to be lower case
-> created entry point, script now runs on command (poetry run app)

LogicAL007 committed:
Fixed the environment variables

- included name and email in the project.toml file
-replace the magic string in the app.py with env variables
-replace the magic string in config.py with env variables

LogicAL007 committed:
Merge branch 'data_ingestion' of https://github.com/Data-Epic/Ghost_Warriors into data_ingestion

LogicAL007 committed:
Update create_table.sql

more descriptive name for temperature
